components/global_header/right_controls/cloud_upgrade_button/index.tsx

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useEffect} from 'react';
```

Comment:
This is missing the code in global header the header to render this component. It also is missing the `getCloudProduct` selector. Since so much has been changing around the cloud limits and selectors part of the code, it would be good to do a spinwick based off this branch and quickly verify that the modal can be opened.